### ISSUE TITLE
Review tutorial

### ISSUE THREAD

**Issue by dmasad** [projectmesa/mesa#176]
Someone take a look at the complete [tutorial draft](https://github.com/projectmesa/mesa/blob/tutorial_update/docs/intro-tutorial.rst) and see how clear you find it. Especially useful would be if someone wants to work through it.

**Comment by dmasad** [comment]
UPDATE: Finally got the branch built in ReadTheDocs, so the nicer-formatted version is available there:
http://mesa.readthedocs.org/en/tutorial_update/intro-tutorial.html

**Comment by paultopia** [comment]
A few things just from reading over the first half of the tutorial.  

1.  It looks like the github link in the first post is slightly more up to date than the readthedocs version.
2.  Out of curiosity, what was the design reason for storing the list of agents as a property of the scheduler rather than of the model directly?  
3.  Re: the todo about `get_cell_list_contents` only taking lists even for single elements, that's a super-easy fix (I'd be inclined just to hack through it by writing a decorator to check for single tuple arguments and wrap them in lists, then just slap it on the relevant methods (it looks like that applies to a number of methods).  Happy to do that myself if you're interested. 
4.  The heat map section is unclear.  The text says "Now let's use matplotlib and numpy to visualize the number of agents residing in each cell."  But the relevant part of the code: 

    wealth_grid = np.zeros((model.grid.width, model.grid.height))
    for cell in model.grid.coord_iter():
        cell_content, x, y = cell
        cell_wealth = len(cell_content)

Seems to imply that what we're visualizing isn't the *number of agents* but the *total wealth* in a cell.  And, because the tutorial doesn't contain much by way of API details for the cell object, there isn't an obvious way to figure out which is actually the case.  (I assume that the cell would have to contain agents, and for wealth we'd really want something like a list comprehension grabbing the wealth of every agent found in the cell... but that's really just an educated guess without digging into the cell code myself.)

**Comment by dmasad** [comment]
Thanks!

1. Which link are you referring to?

2. The design reasoning there was basically maintaining modularity, so that a Schedule didn't require a Model object to go along with it. It's easy enough to add an agent list to the model, or use a decorator to make Model.agents point to Schedule.agents.

3. Yes, that is a super-easy fix. Can you open a ticket for it? And you're welcome / encouraged to take it on! 

4. Good catch! An earlier draft was visualizing the total wealth, and the variable name is a hold-over from there.

**Comment by paultopia** [comment]
Totally.  

1.  I'm referring to this <a href="https://github.com/projectmesa/mesa/blob/tutorial_update/docs/intro-tutorial.rst">this</a> being a little more updated than <a href="http://mesa.readthedocs.org/en/tutorial_update/intro-tutorial.html">this</a> -- I only noticed because there's a comment in the code example right at the start on the github one that answers a question I asked you over twitter re: where the agents go after being created.  

2.  Aaah!  

3.  Sure thing, I'll open an issue and then make it happen.

**Comment by tseminivan** [comment]
My aim was to try out Mesa alongside other ABM tools in a minimum amount of time and effort. I followed this tutorial with the aim of setting up a simple visualization of a grid of agents. I had no interest in the data collection or batch running side quests and therefore skipped those sections.

However I was stumped when my in browser grid was pretty much unresponsive. It responded to reset clicks but it always stopped after a single step afterwards. I also noticed the server returning the message {"type":"end"} each time via tcpdump.

It took me almost an hour of backtracking through the tutorial and exploring the Mesa source code to make sure that I didn't make any mistakes only to discover that there was a crucial 'running' flag in the Model class that had to be initialized for things to start moving.

I subsequently found this crucial piece of information tucked in the Batch Running section  of the tutorial almost as an aside.

Perhaps this should be mentioned earlier in the "Running your first model" section of the tutorial?
